### PR TITLE
Meta: Automatically format Svelte files

### PR DESCRIPTION
**Pull request by fregante**
- Replaces and closes https://github.com/refined-github/refined-github/pull/8082
- #7854 

Compromises. Compromises everywhere.

Pros:

- Svelte components don't suffer from FOUC while the external stylesheet is loaded
- Proper `<style>` usage in Svelte components

Cons:

- CSS selectors with comments still aren't formatted automatically
- Stylelint is not applied to CSS-in-Svelte
	- The suggestion is to use postCSS? sigh https://github.com/stylelint/stylelint/blob/aefbb7afb4c80b5e97acb67139859353987ba05e/docs/migration-guide/to-14.md?plain=1#L76

Future alternatives:

- Try Biome after:
	- CSS-in-Svelte: https://github.com/biomejs/biome/issues/3334
	- https://github.com/biomejs/biome/issues/4575
	- https://github.com/biomejs/biome/issues/4298
- Try dprint after:
	- Someone sends a PR supporting everything we have so far without making the code uglier https://github.com/refined-github/refined-github/issues/7372#issuecomment-2253310879

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,19 @@
...
+		],
+		"singleQuote": true,
+		"overrides": [
+			{
+				"files": "*.svelte",
+				"options": {
+					"parser": "svelte"
+				}
+			}
+		]
```
Comment (fisker) [inline package.json:37]: @fregante Is this `overrides` required? If so, it should be a Prettier bug.

Comment (fregante) [reply package.json:37]: I don't remember, but it's in the docs https://github.com/sveltejs/prettier-plugin-svelte/tree/version-3?tab=readme-ov-file#setup

Comment (fisker) [reply package.json:37]: Thanks.

**Comment by Umeh832** [comment]
Sent from my Galaxy



-------- Original message --------
From: fisker Cheung ***@***.***>
Date: 2/22/25 10:21 AM (GMT+01:00)
To: refined-github/refined-github ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [refined-github/refined-github] Meta: Automatically format Svelte files (PR #8084)


@fisker commented on this pull request.

________________________________

In package.json<https://github.com/refined-github/refined-github/pull/8084#discussion_r1966480219>:

> +             "overrides": [
+                       {
+                               "files": "*.svelte",
+                               "options": {
+                                       "parser": "svelte"
+                               }
+                       }
+               ]


Thanks.

**Pull request merged**